util/ClockAnalysis.py

```
@@ -1,69 +1,52 @@
 StageNameLong = ['fetch', 'decode', 'rename', 'dispatch', 'issue',
                  'arb', 'read', 'execute', 'bypass', 'writeback', 'commit']
 
-
+@lru_cache(maxsize=None)
 def DisAssemble(val):
-    # print(val)
+    if type(val) == str:
+        return val
     hex_val = hex(val).lower()
     command = f'echo "DASM({hex_val})" | spike-dasm'
     asm = subprocess.run(command, shell=True, capture_output=True,
```

Ruff (0.14.8)

`subprocess` call with `shell=True` identified, security issue

(S602)